packages/integration-tests/src/test-projects/data-storage-auth-with-triggers-ts/amplify/func-src/response_generator.ts

```diff
@@ -53,17 +53,35 @@ export const getResponse = async () => {
  */
 const s3RoundTrip = async (): Promise<string> => {
   const filename = 'test.txt';
-  await uploadData({
-    key: filename,
-    data: 'this is some test content',
-    options: {
-      accessLevel: 'guest',
-    },
-  }).result;
+  await retry(
+    () =>
+      uploadData({
+        key: filename,
+        data: 'this is some test content',
+        options: {
+          accessLevel: 'guest',
+        },
+      }).result
+  );
 
-  const downloadResult = await downloadData({
-    key: filename,
-    options: { accessLevel: 'guest' },
-  }).result;
+  const downloadResult = await retry(
+    () =>
+      downloadData({
+        key: filename,
+        options: { accessLevel: 'guest' },
+      }).result
+  );
   return downloadResult.body.text() as Promise<string>;
 };
+
+// executes action and if it throws, executes the action again after a second
+// if the action fails a second time, the error is re-thrown
+const retry = async <T>(action: () => Promise<T>) => {
+  try {
+    return action();
+  } catch (err) {
+    console.log(err);
+    await new Promise((resolve) => setTimeout(resolve, 1000));
+    return action();
+  }
+};
```

Comment:
I think we will definitely end up with a generic retry function somewhere in the e2e harness eventually, but I specifically left this on here because I didn't want to bloat the lambda. Just having this be a single retry with delay keeps the code under test simple